exchange/src/shared/config/config_loader.hpp

```diff
@@ -49,7 +49,11 @@ class Config {
     static const Config&
     get_instance()
     {
-        static Config instance(CONFIG_FILE);
+        const char* config_file_path = std::getenv("NUTC_CONFIG_FILE");
+        if (!config_file_path)
+            config_file_path = DEFAULT_CONFIG_FILE;
+
+        static Config instance(config_file_path);
```

Comment:
nit: we can make config_file_path static to avoid calling getenv every time. Also, this makes it clear that the file path is static for the duration of the program (vs reassignable if `NUTC_CONFIG_FILE` changes)

We call get_instance on the config singleton a lot, so avoiding unnecessary computation is valuable.

```suggestion
        static const char* config_file_path = std::getenv("NUTC_CONFIG_FILE");
        static const char* effective_path = config_file_path ? config_file_path : DEFAULT_CONFIG_FILE;

        static Config instance(effective_path);
```